src/Arcus.Security.Providers.AzureKeyVault/KeyVaultSecretProvider.cs

```diff
@@ -120,6 +149,10 @@ public virtual async Task<Secret> GetSecretAsync(string secretName)
             }
             catch (KeyVaultErrorException keyVaultErrorException)
             {
+                Logger.LogError(keyVaultErrorException,
```

Comment:
This looks like it includes 404, but in that case I don't want to log it every time because maybe it's part of another provider in the secret store?

Reply:
Moved the logging to `else` so we only get it when it's not a 404.